core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -107,6 +107,14 @@ class SparkHadoopUtil extends Logging {
         if (key.startsWith("spark.hadoop.")) {
           hadoopConf.set(key.substring("spark.hadoop.".length), value)
         }
+	// Copy any "fs.swift2d.foo=bar" properties into conf as "fs.swift2d.foo=bar"
```

Comment:
Why filtering, what if we make it a little more generic and pass all properties as is if they are not spark.hadoop.* ?